### PR TITLE
chore: update DevExtreme to 23.1 beta

### PR DESCRIPTION
**Pull request by VasilyStrelyaev**

### CODE REVIEW

File: packages/devextreme-cli/src/utility/latest-versions.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 const packageJson = require('../../package.json');
 module.exports = {
-    'devextreme': '^22.2.4',
-    'devextreme-react': '^22.2.4',
-    'devextreme-vue': '^22.2.4',
+    'devextreme': '23.1.3',
```
Comment (dxvladislavvolkov) [inline packages/devextreme-cli/src/utility/latest-versions.js:3]: Use version with ^

File: packages/devextreme-schematics/src/utility/latest-versions.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // TODO: implement
 export const latestVersions = {
-  'devextreme': '^22.2.4',
-  'devextreme-angular': '^22.2.4',
+  'devextreme': '23.1.3',
```
Comment (dxvladislavvolkov) [inline packages/devextreme-schematics/src/utility/latest-versions.ts:3]: use versions with ^

File: packages/devextreme-schematics/src/install/schema.json [hunk 3]
```diff
@@ -7,7 +7,7 @@
     "dxversion": {
       "type": "string",
       "description": "The DevExtreme version",
-      "default": "^22.2.4"
+      "default": "23.1.3"
```
Comment (dxvladislavvolkov) [inline packages/devextreme-schematics/src/install/schema.json:10]: Versions with ^

File: packages/devextreme-cli/src/templates/react/application/src/components/change-password-form/ChangePasswordForm.tsx [hunk 4]
```diff
@@ -58,6 +58,8 @@ export default function ChangePasswordForm() {
           <RequiredRule message="Password is required" />
           <CustomRule
             message={'Passwords do not match'}
+            // eslint-disable-next-line
+            // @ts-ignore
```
Comment (dxvladislavvolkov) [inline packages/devextreme-cli/src/templates/react/application/src/components/change-password-form/ChangePasswordForm.tsx:62]: Get rid of ts-ignore

File: packages/devextreme-schematics/src/install/schema.json (hunk 3 above)
Comment (VasilyStrelyaev) [reply packages/devextreme-schematics/src/install/schema.json]: fixed

File: packages/devextreme-schematics/src/utility/latest-versions.ts (hunk 2 above)
Comment (VasilyStrelyaev) [reply packages/devextreme-schematics/src/utility/latest-versions.ts]: fixed

File: packages/devextreme-cli/src/utility/latest-versions.js (hunk 1 above)
Comment (VasilyStrelyaev) [reply packages/devextreme-cli/src/utility/latest-versions.js]: fixed

File: packages/devextreme-cli/src/templates/react/application/src/components/change-password-form/ChangePasswordForm.tsx (hunk 4 above)
Comment (VasilyStrelyaev) [reply packages/devextreme-cli/src/templates/react/application/src/components/change-password-form/ChangePasswordForm.tsx:62]: @iBat was it required?

File: .github/workflows/applications.yml [hunk 5]
```diff
@@ -40,38 +41,14 @@ jobs:
...
-        path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
-        key: ${{ runner.os }}-yarn-${{ matrix.NODE }}-${{ hashFiles('**/yarn.lock') }}
-        restore-keys: |
-          ${{ runner.os }}-yarn-${{ matrix.NODE }}
-          ${{ runner.os }}-yarn-
-
-    - name: Disable taobao registry
-      if: ${{ startsWith( matrix.APPROACH, 'vue') }}
-      run: |
-        echo {\"useTaobaoRegistry\": false} > ${{ env.VUE_CLI_CONFIG_PATH }}
```
Comment (OlegKipchatov) [inline .github/workflows/applications.yml:66]: In the past, Vue CI had many failures when installing npm packages, and the `useTaobaoRegistry` support resolved this issue. Perhaps we shouldn't remove this option?

File: packages/devextreme-cli/package.json [hunk 6]
```diff
@@ -30,7 +30,7 @@
     "test-template": "node ./testing/testing.js",
     "test-dev": "cross-env TEST_MODE=dev jest",
     "test-themebuilder": "jest --config=./jest-themebuilder.config.js --runInBand",
-    "docker:build": "docker build --rm -f \"testing/docker/Dockerfile\" -t devextreme-cli:latest \"testing/docker\"",
+    "docker:build": "docker build --rm -f \"testing/docker/Dockerfile\" -t devextreme-cli:latest \"testing/docker\" --progress=plain",
```
Comment (OlegKipchatov) [inline packages/devextreme-cli/package.json:33]: Are we sure we still need docker support?

File: packages/devextreme-cli/src/applications/application.vue.js [hunk 7]
```diff
@@ -56,7 +56,7 @@ const preparePackageJsonForTemplate = (appPath, appName, version) => {
 async function createVueApp(name, templateOptions) {
     const { version } = templateOptions;
     const toolingVersion = extractToolingVersion(templateOptions);
-    const argList = ['-p', `@vue/cli${toolingVersion}`, 'vue', 'create', name];
+    const argList = ['-p', `@vue/cli${toolingVersion}`, 'vue', 'create', name, '--registry', 'https://registry.npmjs.org/'];
```
Comment (OlegKipchatov) [inline packages/devextreme-cli/src/applications/application.vue.js:59]: Are we certain that we want to discontinue support for customers to install packages from their local npm repositories?

File: .github/workflows/applications.yml (hunk 5 above)
Comment (VasilyStrelyaev) [reply .github/workflows/applications.yml:66]: Please @iBat reply

File: packages/devextreme-cli/package.json (hunk 6 above)
Comment (VasilyStrelyaev) [reply packages/devextreme-cli/package.json:33]: Please @iBat reply

File: packages/devextreme-cli/src/applications/application.vue.js (hunk 7 above)
Comment (VasilyStrelyaev) [reply packages/devextreme-cli/src/applications/application.vue.js:59]: Please @iBat reply

File: .github/workflows/applications.yml (hunk 5 above)
Comment (iBat) [reply .github/workflows/applications.yml:66]: This was achieved by pining registry in [packages/devextreme-cli/src/applications/application.vue.js](https://github.com/DevExpress/devextreme-cli/pull/819/files/fe475ab8065ebfe910b2875aecf753a190bc0c6b#diff-16df00d52a46c198d2c7ca0e74aac912be612bbccae48d53042202ba8e452741)
so this step is unnecessary. Btw now it doesn't work.

File: packages/devextreme-cli/package.json (hunk 6 above)
Comment (iBat) [reply packages/devextreme-cli/package.json:33]: Docker is not currently used in the CI/CD workflow. But probably still needed in development.

File: packages/devextreme-cli/src/applications/application.vue.js (hunk 7 above)
Comment (iBat) [reply packages/devextreme-cli/src/applications/application.vue.js:59]: I have found no other way to disable the use of the taobao registry. The "useTaobaoRegistry" option doesn't seem to work.

File: packages/devextreme-cli/src/templates/react/application/src/components/change-password-form/ChangePasswordForm.tsx (hunk 4 above)
Comment (iBat) [reply packages/devextreme-cli/src/templates/react/application/src/components/change-password-form/ChangePasswordForm.tsx:62]: This is where the TS linter error is without ts-ignore.

Comment (VasilyStrelyaev) [reply packages/devextreme-cli/src/templates/react/application/src/components/change-password-form/ChangePasswordForm.tsx]: Okay, I will fix the cause of the fail

File: packages/devextreme-cli/src/applications/application.vue.js (hunk 7 above)
Comment (OlegKipchatov) [reply packages/devextreme-cli/src/applications/application.vue.js:59]: Maybe we could turn on this only for testing?

File: packages/devextreme-cli/package.json (hunk 6 above)
Comment (VasilyStrelyaev) [reply packages/devextreme-cli/package.json:33]: Since it's now paid, it makes sense to remove it completely.

**Review by OlegKipchatov — DISMISSED** [review]

**Review by dxvladislavvolkov — APPROVED** [review]

**Review by dxvladislavvolkov — APPROVED** [review]

**Pull request merged**